compiler/transforms/snax_copy_to_dma.py

```diff
@@ -28,6 +28,42 @@
 from compiler.ir.tsl import TiledStridedLayout
 
 
+def get_total_size_op(source: SSAValue):
+    assert isinstance(source.type, MemRefType)
+
+    # step 1: extract size information to calculate total size
+    # this is done by multiplying all the shape dimensions
+    ops_to_insert = []
+    total_size_op = None
+    for dim in range(source.type.get_num_dims()):
+        const_op = Constant.from_int_and_width(dim, IndexType())
+        ops_to_insert.append(const_op)
+        dim_op = Dim.from_source_and_index(source, const_op.result)
+        ops_to_insert.append(dim_op)
+        if total_size_op is None:
+            total_size_op = dim_op
+        else:
+            total_size_op = Muli(total_size_op.result, dim_op.result, IndexType())
+            ops_to_insert.append(total_size_op)
```

Comment:
Can you move this to a utility function (maybe separate PR) ? It is used in a lot of places

Reply:
Ok, but where is it used elsewhere?

Reply:
oh wait I'm probably overreacting :dango: